joblib/numpy_pickle_utils.py

```diff
@@ -0,0 +1,449 @@
+    the magic number.
+
+    Parameters
+    ----------
+    file_handle: file_like
+
+    """
+    magic = file_handle.read(len(_ZFILE_PREFIX))
+    # Pickling needs file-handles at the beginning of the file
+    file_handle.seek(0)
+    return magic
+
+
+def _check_filetype(filename, magic):
+    """Utility function opening the right fileobject from a filename.
+
+    The magic number is used to choose between the type of file object to open:
+    * regular file object (default)
+    * gzip file object is magic matches _GZIP_PREFIX
+
+    Parameters
+    ----------
+    filename: str
+        The full path string of the file
+    magic: bytes
+        The bytes representation of the magic number of the file
+
+    Returns
+    -------
+        a file like object
+
+    """
+    if magic == _GZIP_PREFIX:
+        return gzip_file_factory(filename)
+
+    return open(filename, 'rb')
+
+
+def _open_memmap(filename, array_offset=0, mode='r+', dtype=None, shape=None,
+                 fortran_order=False, version=None):
+    """
+    Open a .npy file as a memory-mapped array.
+
+    This may be used to read an existing file or create a new one.
+
+    Parameters
+    ----------
+    filename : str
+        The name of the file on disk.  This may *not* be a file-like
+        object.
+    array_offset : int
+        The offset in the file object where the array is serialized
+    mode : str, optional
+        The mode in which to open the file; the default is 'r+'.  In
+        addition to the standard file modes, 'c' is also accepted to mean
+        "copy on write."  See `memmap` for the available mode strings.
+    dtype : data-type, optional
+        The data type of the array if we are creating a new file in "write"
+        mode, if not, `dtype` is ignored.  The default value is None, which
+        results in a data-type of `float64`.
+    shape : tuple of int
+        The shape of the array if we are creating a new file in "write"
+        mode, in which case this parameter is required.  Otherwise, this
+        parameter is ignored and is thus optional.
+    fortran_order : bool, optional
+        Whether the array should be Fortran-contiguous (True) or
+        C-contiguous (False, the default) if we are creating a new file in
+        "write" mode.
+    version : tuple of int (major, minor) or None
+        If the mode is a "write" mode, then this is the version of the file
+        format used to create the file.  None means use the oldest
+        supported version that is able to store the data.  Default: None
+
+    Returns
+    -------
+    marray : memmap
+        The memory-mapped array.
+
+    Raises
+    ------
+    ValueError
+        If the data or the mode is invalid.
+    IOError
+        If the file is not found or cannot be opened correctly.
+
+    See Also
+    --------
+    memmap
+
+    Note
+    ----
+    Monkey patched version of open_memmap function taken from
+    numpy.lib.format.py file.
```

Comment:
Either use the fully-qualified function name e.g. numpy.lib.format.open_memmap or the file with slashes e.g. open_memmap from numpy/lib/format.py.

Also mention the numpy version you took it from.